src/omotes_simulator_core/adapter/transforms/mappers.py

```diff
@@ -131,32 +108,36 @@ def to_entity(self, network: Network) -> Tuple[List[AssetAbstract], List[Junctio
         This method first converts all assets into a list of assets.
         Next to this a list of Junctions is created. This is then used
         to create the Heatnetwork object.
-        :param Network network: network to add the compenents to.
+        :param Network network: network to add the components to.
         :return: (List[AssetAbstract], List[Junction]), tuple of list of assets and junctions.
         """
         # TODO: This method requires a clean-up!
-        py_assets_list = []
-        py_joint_dict = {}
-        for esdl_asset in self.esdl_object.get_all_assets_of_type("asset"):
-            # Esdl Junctions need to be skipped for now, are added later.
-            if isinstance(esdl_asset.esdl_asset, esdl_junction):
-                py_joint_dict[esdl_asset.esdl_asset.id] = [
-                    *self.esdl_object.get_connected_assets(esdl_asset.esdl_asset.id, Port.In),
-                    *self.esdl_object.get_connected_assets(esdl_asset.esdl_asset.id, Port.Out),
-                ]
-            else:
-                py_assets_list.append(EsdlAssetMapper().to_entity(esdl_asset))
-                py_assets_list[-1].add_physical_data(esdl_asset=esdl_asset)
-                network.add_existing_asset(py_assets_list[-1].solver_asset)
+        py_assets_list = self._convert_assets(network)
```

Comment:
Why would you define _create_junctions and _convert_assets? Could you rename to have the same names?

Reply:
No, create_node really creates nodes which do not exist, while convert asset is converting the esdl assets to our own assets.